modules/services/swaync.nix

```diff
@@ -0,0 +1,154 @@
+        description = ''
+          Systemd target to bind to.
+        '';
+      };
+    };
+
+    style = lib.mkOption {
+      type = lib.types.nullOr (lib.types.either lib.types.path lib.types.lines);
+      default = null;
+      description = ''
+        CSS style of the bar.
+        </para>
+        <para>
+        See <link xlink:href="https://github.com/ErikReider/SwayNotificationCenter/blob/main/src/style.css"/>
+        for the documentation.
+        </para>
+        <para>
+        If the value is set to a path literal, then the path will be used as the css file.
+      '';
+      example = ''
+        .notification-row {
+          outline: none;
+        }
+
+        .notification-row:focus,
+        .notification-row:hover {
+          background: @noti-bg-focus;
+        }
+
+        .notification {
+          border-radius: 12px;
+          margin: 6px 12px;
+          box-shadow: 0 0 0 1px rgba(0, 0, 0, 0.3), 0 1px 3px 1px rgba(0, 0, 0, 0.7),
+            0 2px 6px 2px rgba(0, 0, 0, 0.3);
+          padding: 0;
+        }
+      '';
+    };
+
+    settings = lib.mkOption {
+      type = jsonFormat.type;
+      default = { };
+      description = ''
+        Configuration written to <filename>$XDG_CONFIG_HOME/swaync/config.json</filename>.
+        </para>
+        <para>
+        See <link xlink:href="https://github.com/ErikReider/SwayNotificationCenter/blob/main/src/configSchema.json"/>
+      '';
+      example = lib.literalExpression ''
+        {
+          positionX = "right";
+          positionY = "top";
+          layer = "overlay";
+          control-center-layer = "top";
+          layer-shell = true;
+          cssPriority = "application";
+          control-center-margin-top = 0;
+          control-center-margin-bottom = 0;
+          control-center-margin-right = 0;
+          control-center-margin-left = 0;
+          notification-2fa-action = true;
+          notification-inline-replies = false;
+          notification-icon-size = 64;
+          notification-body-image-height = 100;
+          notification-body-image-width = 200
+        };
+      '';
+    };
+  };
+
+  config = lib.mkIf cfg.enable {
+    # at-spi2-core is to minimize journalctl noise of:
+    # "AT-SPI: Error retrieving accessibility bus address: org.freedesktop.DBus.Error.ServiceUnknown: The name org.a11y.Bus was not provided by any .service files"
+    home.packages = [ cfg.package pkgs.at-spi2-core ];
+
+    xdg.configFile =
+      let onChange = "${pkgs.procps}/bin/pkill -u $USER -USR2 swaync || true";
+      in {
+        "swaync/config.json" = {
+          source = configFile;
+          inherit onChange;
+        };
+        "swaync/style.css" = lib.mkIf (cfg.style != null) {
+          source =
+            if builtins.isPath cfg.style || lib.isStorePath cfg.style then
+              cfg.style
+            else
+              pkgs.writeText "swaync/style.css" cfg.style;
+          inherit onChange;
+        };
```

Comment:
I think you can remove the onChange stuff. Just add

```
Unit.X-Restart-Triggers = [
  "${config.xdg.configFile."swaync/config.json".source}"
  "${config.xdg.configFile."swaync/style.css".source}"
]
```

to the service definition.

Reply:
There's a couple of other places where this improvement could be made.